controlplane/src/core/repositories/GraphCompositionRepository.ts

```diff
@@ -0,0 +1,50 @@
+import { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
+import { eq } from 'drizzle-orm';
+import * as schema from '../../db/schema.js';
+import { graphCompositions, schemaVersion, targets } from '../../db/schema.js';
+import { GraphCompositionDTO, SubgraphDTO } from '../../types/index.js';
+
+export class GraphCompositionRepository {
+  constructor(private db: PostgresJsDatabase<typeof schema>) {}
+
+  public async addComposition(input: { fedGraphSchemaVersionId: string; subgraphSchemaVersionIds: string[] }) {
+    if (input.subgraphSchemaVersionIds.length > 0) {
```

Comment:
Matter of style but I'd always early return instead of wrapping the code with the if case. It becomes easier to read.